ansible_collections/f5networks/f5_modules/plugins/modules/bigip_ssl_key_cert.py

```diff
@@ -0,0 +1,815 @@
+author:
+  - Nitin Khanna (@nitinthewiz)
+'''
+
+EXAMPLES = r'''
+- name: Create a ...
+  bigip_ssl_key_cert:
+    name: foo
+    provider:
+      password: secret
+      server: lb.mydomain.com
+      user: admin
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+param1:
+  description: The new param1 value of the resource.
+  returned: changed
+  type: bool
+  sample: true
+param2:
+  description: The new param2 value of the resource.
+  returned: changed
+  type: str
+  sample: Foo is bar
+'''
+
+import hashlib
+import os
+import re
+
+from ansible.module_utils.basic import (
+    AnsibleModule, env_fallback
+)
+
+from ..module_utils.bigip import F5RestClient
+from ..module_utils.common import (
+    F5ModuleError, AnsibleF5Parameters, transform_name, f5_argument_spec, fq_name
+)
+from ..module_utils.icontrol import (
+    TransactionContextManager, upload_file
+)
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+
+def merge_two_dicts(x, y):
+    z = x.copy()
+    z.update(y)
+    return z
+
+
+class Parameters(AnsibleF5Parameters):
+    download_path = '/var/config/rest/downloads'
+
+    api_map = {
+        'sourcePath': 'source_path',
+        'issuerCert': 'issuer_cert',
+    }
+
+    api_attributes = [
+        'passphrase',
+        'sourcePath',
+        'issuerCert',
+    ]
+
+    returnables = [
+        'filename',
+        'checksum',
+        'source_path',
+        'issuer_cert',
+    ]
+
+    updatables = [
+        'checksum',
+        'key_checksum',
+        'cert_checksum',
+        'content',
+        'issuer_cert',
+        'source_path',
+    ]
+
+
+class ApiParameters(Parameters):
+    @property
+    def checksum(self):
+        if self._values['checksum'] is None:
+            return None
+        pattern = r'SHA1:\d+:(?P<value>[\w+]{40})'
+        matches = re.match(pattern, self._values['checksum'])
+        if matches:
+            return matches.group('value')
+        else:
+            return None
+
+    @property
+    def key_filename(self):
+        if self._values['name'] is None:
+            return None
+        if not self._values['name'].endswith('.key'):
+            return None
+        return self._values['name']
+
+    @property
+    def key_source_path(self):
+        if self.key_filename is None:
+            return None
+        if self._values['sourcePath'] is None:
+            return None
+        else:
+            return self._values['sourcePath']
+
+    @property
+    def cert_filename(self):
+        if self._values['name'] is None:
+            return None
+        if not self._values['name'].endswith('.crt'):
+            return None
+        return self._values['name']
+
+    @property
+    def cert_source_path(self):
+        if self.cert_filename is None:
+            return None
+        if self._values['sourcePath'] is None:
+            return None
+        else:
+            return self._values['sourcePath']
```

Comment:
how is this any different from key_source_path? also you already do api key map here: https://github.com/F5Networks/f5-ansible/pull/1814/commits/5f31b3d05e6bb0c065628f943294fa7307cc3099#diff-b157170dfeade030fbc34107ad88befeR135

so the value would be source_path not sourcePath

Reply:
I need to have them specified separately because otherwise when both are specified. Perhaps I need to do to this the same I've done with key_checksum and cert_checksum, and remove these.